3-projects/modules/project_subnet/main.tf

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  subnet_data = [
+    {
+      subnet_name           = var.application_name
+      subnet_ip             = var.ip_cidr_range
+      subnet_region         = var.default_region
+      subnet_private_access = var.enable_private_access
+      subnet_flow_logs      = var.enable_vpc_flow_logs
+      description           = var.application_name
+    }
+  ]
+  subnet           = var.enable_networking == true ? local.subnet_data : []
+  secondary_ranges = var.enable_networking == true ? var.secondary_ranges : []
+  subnet_self_link = var.enable_networking == true ? [for subnet in module.project_subnet.subnets : replace(subnet.self_link, "https://www.googleapis.com/compute/v1/", "")] : []
```

Comment:
As discussed, can we add an output for the subnet self link and instructions on how to update project factory to refer to it?